common/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/network/GenericNetworkQuery.kt

```diff
@@ -31,3 +31,49 @@ GetPasswordTokenInteractor) : PaginationOffsetLimitQuery(identifier ?: "$offset-
 class OAuthPasswordObjectQuery<T>(value: T, val getPasswordTokenInteractor: GetPasswordTokenInteractor) : ObjectQuery<T>(value),
     OAuthClientQuery
 
+abstract class GenericNetworkQuery(
+    val path: String,
+    val params: List<Pair<String, String>>?,
+    val headers: List<Pair<String, String>>?,
+    cacheKey: String? = null) : KeyQuery(cacheKey ?: path)
+
+abstract class GenericOAuthNetworkQuery(
+    val getPasswordTokenInteractor: GetPasswordTokenInteractor,
+    path: String,
+    params: List<Pair<String, String>>? = emptyList(),
+    headers: List<Pair<String, String>>? = emptyList(),
+    cacheKey: String? = null) : GenericNetworkQuery(path, params, headers, cacheKey)
+
+// PUT Create JavaDoc
+abstract class GenericIdNetworkQuery(
+    val id: Int,
```

Comment:
ID should not be an Int as it could be many things, for example, a UUID, etc. I think we need to use a generic ID only.

Reply:
Change to generic type

Reply:
Handle if not int or string then throw exception